packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -407,6 +409,26 @@ function computeStylesheets(vm: VM, ctor: LightningElementConstructor) {
     return null;
 }
 
+function warnOnStylesheetsMutation(ctor: LightningElementConstructor) {
+    if (process.env.NODE_ENV !== 'production') {
+        let stylesheets = ctor.stylesheets;
+        defineProperty(ctor, 'stylesheets', {
+            enumerable: true,
+            configurable: true,
+            get() {
+                return stylesheets;
+            },
+            set(newValue) {
+                logError(
+                    'Dynamically setting the stylesheets static property on a LightningElementConstructor ' +
+                        'will not affect the stylesheets injected.'
+                );
```

Comment:
```suggestion
                logWarnOnce(
                    'Dynamically setting the "stylesheets" static property ' +
                        'will not affect the stylesheets injected.'
                );
```

```suggestion
                    'Dynamically setting the "stylesheets" static property ' +
                        'will not affect the stylesheets injected.'
```

I would avoid putting `LightningElementConstructor` because I don't think most developers know what this means.

I would also use `logWarnOnce` because 1) this seems more warning-level than error-level, and 2) it would probably be annoying to see this message more than once in the console.

Reply:
On top of the `logWarnOnce` could we also add the constructor name to quickly identify which is the incriminated component?